channeldb/messagestore.go

```diff
@@ -0,0 +1,153 @@
+package channeldb
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	"github.com/boltdb/bolt"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/roasbeef/btcd/wire"
+)
+
+var (
+	// basePeerBucketKey is the base key for generating peer bucket keys.
+	// Peer bucket stores top-level bucket that maps: index -> code || msg
+	// Concatenating the message code to the stored data allows the db logic
+	// to properly parse the wire message without trial and error or needing
+	// an additional index.
+	basePeerBucketKey = []byte("peermessagesstore")
+)
+
+// MessagesStore represents the boltdb storage for messages inside
+// retransmission sub-system.
+type MessagesStore struct {
+	// id is a unique identificator of peer.
```

Comment:
I re-thinked the meaning of this field a bit due to the recent changes in discovery PR. I think it would be better to keep `id` as not something coupled with peer at all, but mention that usually it is compressed pub key.